### DISCUSSION TITLE
const puppeteer = require('puppeteer'); ReferenceError: require is not defined in ES module scope

### DISCUSSION THREAD

**Discussion by aladaghlou** [spatie/browsershot#744] [General]
Hi.
I installed node and puppeteer in debian 11. Then install laravel 10 and browsershot.

```
use Illuminate\Support\Facades\Route;
use Spatie\Browsershot\Browsershot;

Route::get('/', function () {
    Browsershot::url("https://facebook.com")
    ->noSandbox()
    ->showBackground()
    ->save("aaa.pdf");
});
```

when i run page i got this error:

```
Error Output:
================
file:///var/www/html/example-pdf/vendor/spatie/browsershot/bin/browser.js:1
const puppeteer = require('puppeteer');
                  ^
ReferenceError: require is not defined in ES module scope, you can use import instead
This file is being treated as an ES module because it has a '.js' file extension and '/var/www/html/example-pdf/package.json' contains "type": "module". To treat it as a CommonJS script, rename it to use the '.cjs' file extension.
    at file:///var/www/html/example-pdf/vendor/spatie/browsershot/bin/browser.js:1:19
    at ModuleJob.run (node:internal/modules/esm/module_job:194:25)

Node.js v18.16.0
```

**Comment by fvanzeijl** [comment]
Hi,
it caused by the `type: module` in your package.json.

So you can change it to `commonjs`. The downside is that if you are using laravel-vite with postcss the frontend-compiling does not work anymore becuase it only works with type: module.

In that case you can rename your `postcss.config.js` tp `postcss.config.mjs` and remove the type-param from your package.json.

Hope that helps.

**Comment by aladaghlou** [comment]
> Hi, it caused by the `type: module` in your package.json.
> 
> So you can change it to `commonjs`. The downside is that if you are using laravel-vite with postcss the frontend-compiling does not work anymore becuase it only works with type: module.
> 
> In that case you can rename your `postcss.config.js` tp `postcss.config.mjs` and remove the type-param from your package.json.
> 
> Hope that helps.

Excellent this is worked for me 👍❤

**Comment by osbre** [comment]
Fixed now because my PR got accepted:
https://github.com/spatie/browsershot/pull/740

Make sure to update the package.
